File: 2178-maximum-split-of-positive-even-integers/2178-maximum-split-of-positive-even-integers.cpp
class Solution {
public:
    vector<long long> maximumEvenSplit(long long s) {
        if(s%2!=0)

        {
            return {};
        }
        else
        {
            vector<long long>vec;
            long long sum=0;
            for(long long i=2;i<=s;i+=2)
            {
                sum+=i;
                long long tem=s-sum;
                
                if(tem<=i)
                {
                    vec.push_back(i+tem);
                    return vec;
                }
                else
                {
                    vec.push_back(i);
                }
                if(sum==s)
                {
                    return vec;
                }
            }
            
            return vec;
        }
    }
};

File: 2178-maximum-split-of-positive-even-integers/2178-maximum-split-of-positive-even-integers.py
class Solution:
    def maximumEvenSplit(self, s: int) -> List[int]:
        if(s%2!=0):
            return {};
        else:
            vec=[];
            sum=0;
            for i in range (2,s+1,2):
                sum+=i;
                temp=s-sum;
                if(temp<=i):
                    vec.append(i+temp);
                    return vec;
                else:
                    vec.append(i);
                if(sum==s):
                    return vec;
        return vec;
        